File: Python3/2032_two_out_of_three.py
# https://leetcode.cn/problems/two-out-of-three/

class Solution:
    def twoOutOfThree(self, nums1, nums2, nums3):
        """
        Inputs:
            nums1, nums2, nums3: list[int]
        Outputs:
            res: list[int]
        """
        s1, s2, s3 = set(nums1), set(nums2), set(nums3)
        inter1 = s1.intersection(s2)
        inter2 = s2.intersection(s3)
        inter3 = s3.intersection(s1)
        res = list(inter1.union(inter2, inter3))
        return res

File: Cpp/2032_two_out_of_three.cpp
// https://leetcode.cn/problems/two-out-of-three/

class Solution {
public:
    vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
        set<int> s1(nums1.begin(), nums1.end());
        set<int> s2(nums2.begin(), nums2.end());
        set<int> s3(nums3.begin(), nums3.end());
        set<int> inter1, inter2, inter3;
        set_intersection(s1.begin(), s1.end(), s2.begin(), s2.end(),
            inserter(inter1, inter1.begin()));
        set_intersection(s1.begin(), s1.end(), s3.begin(), s3.end(),
            inserter(inter2, inter2.begin()));
        set_intersection(s2.begin(), s2.end(), s3.begin(), s3.end(),
            inserter(inter3, inter3.begin()));
        inter1.insert(inter2.begin(), inter2.end());
        inter1.insert(inter3.begin(), inter3.end());
        vector<int> res(inter1.begin(), inter1.end());
        return res;
    }
};
